common/meta/types/src/app_error.rs

```diff
@@ -283,39 +283,43 @@ impl ShareAlreadyExists {
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, thiserror::Error)]
-#[error("ShareAccountAlreadyExists: {share_name} while {context}")]
-pub struct ShareAccountAlreadyExists {
+#[error("ShareAccountsAlreadyExists: {share_name} while {context}")]
+pub struct ShareAccountsAlreadyExists {
     share_name: String,
-    account: String,
+    accounts: Vec<String>,
     context: String,
 }
 
-impl ShareAccountAlreadyExists {
+impl ShareAccountsAlreadyExists {
     pub fn new(
         share_name: impl Into<String>,
-        account: impl Into<String>,
+        accounts: impl Into<Vec<String>>,
```

Comment:
Better using `Iterator<Item=String>` or `&[String]` here, IMHO.

`Into<String>` would be a good abstraction for a single account, but `Into<Vec<String>>` is not very common I think.